### PR TITLE
Retry transaction for read write keeping same transaction ID over retries

### PR DESCRIPTION
**Pull request by chemelnucfin**

### CODE REVIEW

**Comment by chemelnucfin** [comment]
Somehow the tests are different locally and on circle ci.  Will look into it.

File: datastore/google/cloud/datastore/transaction.py [hunk 1]
```diff
@@ -262,3 +279,28 @@ def put(self, entity):
             raise RuntimeError("Transaction is read only")
         else:
             super(Transaction, self).put(entity)
+
+
+def _sleep(current_sleep, max_sleep=30.0, multiplier=2.0):
```
Comment (dhermes) [inline datastore/google/cloud/datastore/transaction.py]: You shouldn't need to implement a retry here, it should be in `api_core`

File: datastore/google/cloud/datastore/batch.py [hunk 2]
```diff
@@ -270,9 +270,10 @@ def commit(self):
             raise ValueError('Batch must be in progress to commit()')
 
         try:
-            self._commit()
+            commit_response_pb = self._commit()
         finally:
             self._status = self._FINISHED
+        return commit_response_pb
```
Comment (dhermes) [inline datastore/google/cloud/datastore/batch.py]: Why did you add this? It doesn't seem like it's used in `transaction.py`?

**Comment by dhermes** [comment]
@chemelnucfin The code in this PR doesn't seem to match the description. Can you clarify?

**Comment by chemelnucfin** [comment]
I don't think this is needed any more.
